### PR TITLE
Order history page for users

### PR DESCRIPTION
**Pull request by anamariamilcu**
I added a page where users can see all the books they borrowed in the past, without having duplicate books.

### CODE REVIEW

File: src/main/java/com/google/sps/data/OrderHandler.java [hunk 1]
```diff
@@ -187,6 +186,10 @@ public void placeOrder(String uid)
     makeOrders(uid, Point.fromAddress((String) userEntity.getProperty("address")),
           (ArrayList<String>) cartEntity.getProperty("books"));
 
+    // Add the ordered books to the ordering history of the user.
+    OrderHistory orderHistory = new OrderHistory();
```
Comment (alexcepoi) [inline src/main/java/com/google/sps/data/OrderHandler.java:221]: nit: "OrderHistory orderHistory;" should be enough here, as there's not much here that requires allocating on the heap.

https://dzone.com/articles/stack-vs-heap-understanding-java-memory-allocation

File: src/test/javascript/testBookScript.js [hunk 2]
```diff
@@ -81,3 +81,13 @@ describe('Testing common.js - parseQueryString', () => {
     expect(result).toEqual({});
   });
 });
+
+describe('Testing common.js - getBookJSON', () => {
+  it('getting the correct info about a book', async () => {
+    const result = await getBookJSON('OBM3AAAAIAAJ');
```
Comment (alexcepoi) [inline src/test/javascript/testBookScript.js:87]: In general, unit tests should be hermetic, that means they don't require external dependencies. This unit test is actually calling into the real Google Books API, so any issue with the API availability will result in your tests flaking.

What you would generally do here is mock the Google Books API. That's like saying, I don't care what the real Google Books API actually returns, but assuming it returns this, getBookJSON will behave in this way. Not sure what's the best way to do this, but  http://www.wheresrhys.co.uk/fetch-mock/ seems promising.

That said, this function doesn't really do anything other than a fetch request, so it wouldn't actually be testing anything worth testing. This trick could be useful though if you want to be able to test any sort of processing of the google books api response.

File: src/test/java/com/google/sps/data/OrderHistoryTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+  @Test
+  public void testAddingDuplicateToUserHistory() {
+    orderHistory.addBooksToOrderHistory("1234", Arrays.asList("book1", "book2"));
+    Assert.assertEquals(Arrays.asList("book1", "book2"), orderHistory.getUserOrderHistory("1234"));
+    orderHistory.addBooksToOrderHistory("1234", Arrays.asList("book5", "book2"));
+    Assert.assertEquals(Arrays.asList("book1", "book2", "book5"), orderHistory.getUserOrderHistory("1234"));
+  }
+
+}
```
Comment (alexcepoi) [inline src/test/java/com/google/sps/data/OrderHistoryTest.java]: newline at end of file.

**Review by ak47na — APPROVED** [review]

File: src/main/webapp/common.js [hunk 4]
```diff
@@ -305,3 +305,20 @@ async function addMenu() {
...
+ * book with given ID to be able to print the title and author of the book on
+ * the shopping cart page.
+ * @param {String} id 
+ */
+
+async function getBookJSON(id) {
+  // To get results from UK publishing houses.
+  const COUNTRY = 'UK';
+  let URL = `https://www.googleapis.com/books/v1/volumes/${id}`;
+  URL += `?country=${COUNTRY}`;
```
Comment (alexcepoi) [inline src/main/webapp/common.js]: What's with all the variables and concatenation? Why not have a single string?

**Review by alexcepoi — APPROVED** [review]
Nice clean PRs.

Good job, Ana-Maria!

File: src/main/java/com/google/sps/data/OrderHandler.java (hunk 1 above)
Comment (anamariamilcu) [reply src/main/java/com/google/sps/data/OrderHandler.java:221]: I get an error if I don't initialize it?

File: src/test/javascript/testBookScript.js (hunk 2 above)
Comment (anamariamilcu) [reply src/test/javascript/testBookScript.js:87]: I see, I will just give up on it then.

File: src/test/java/com/google/sps/data/OrderHistoryTest.java (hunk 3 above)
Comment (anamariamilcu) [reply src/test/java/com/google/sps/data/OrderHistoryTest.java]: Done

File: src/main/webapp/common.js (hunk 4 above)
Comment (anamariamilcu) [reply src/main/webapp/common.js]: Done

**Pull request merged**